src/output_gstreamer.h

```diff
@@ -25,6 +25,73 @@
 #ifndef _OUTPUT_GSTREAMER_H
 #define _OUTPUT_GSTREAMER_H
 
-extern struct output_module gstreamer_output;
+#include <gst/gst.h>
 
-#endif /*  _OUTPUT_GSTREAMER_H */
+#include "output_module.h"
+
+class GstreamerOutput : public OutputModule,
+                        public OutputModuleFactory<GstreamerOutput> {
+ public:
+  struct Options : public OutputModule::Options {
```

Comment:
mm, so this dance below to cast the options to the specific type because we happen to know that we get the right type when our Initialize() is called is brittle and breaks abstraction. This is something we might ponder in a later change how this can be improved. Ok for now.